R/importTaxpasta.R

```diff
@@ -0,0 +1,116 @@
+#' Import taxpasta-specific BIOM results to `TreeSummarizedExperiment`
+#'
+#' @param filename A character vector with one element that denotes the file
+#'   path to a BIOM file.
+#'
+#' @return A [TreeSummarizedExperiment::TreeSummarizedExperiment()] object.
+#' @export
+#'
+#' @examples
+#' tse <- importTaxpasta("profiles.biom")
+#'
+#' @importFrom rhdf5 h5read
+#' @importFrom SummarizedExperiment rowData
+#' @importFrom SingleCellExperiment altExps
+importTaxpasta <- function(filename) {
+    # Validate the input.
+    if (!.is_non_empty_string(filename)) {
+        stop("'filename' must be a single character value.", call. = FALSE)
+    }
+    if (!file.exists(filename)) {
+        stop(filename, " not found.", call. = FALSE)
+    }
+
+    # We read our own HDF5 array to later be able to read observation group
+    # metadata, which [biomformat::read_biom()] currently doesn't do.
+    raw <- h5read(filename, "/", read.attributes = TRUE)
+    biom <- .createBiom(raw)
+
+    # Without taxonomic information, we return a simple TreeSE.
+    if (is.null(raw$observation$`group-metadata`$ranks)) {
+        warning("The BIOM file does not contain taxonomy information; unable to generate a taxonomic tree.")
```

Comment:
Split long lines to multiple lines. 80 characters is the max (Bioconductor guidelines). Run BiocCheck::BiocCheck() and fix things related to added lines

Reply:
Also add warning(call. = FALSE)

Reply:
Also, related to coding-style I would rather have one return() call per function. So instead of returning the object inside the if, you would add the taxonomy.

But that is more "matter of taste" kind of thing. (Sorry for little bit cryptic review, it is late here, and I wanted to comment so that I don't forget to do so)

Reply:
I've split lines and added `call. = FALSE`.

I disagree with you preferring a single return. The early return is a form of a guard clause which is good practice. It makes the code easier to read and reason about. See, for example, https://en.wikipedia.org/wiki/Guard_(computer_science) and https://www.oreilly.com/library/view/tidy-first/9781098151232/ch01.html. Maybe a relevant excerpt from the latter reference:

> (But what about MuLTipLe ReTuRns? The “rule” about having a single return for a routine came from the days of FORTRAN, where a single routine could have multiple entry and exit points. It was nearly impossible to debug such code. You couldn’t tell what statements were executed. Code with guard clauses is easier to analyze because the preconditions are explicit.)